<launch>

  <!-- args corresponding to environment variables -->
  <arg name="UAV_NAME" default="$(env UAV_NAME)" />
  <arg name="PROFILER" default="$(optenv PROFILER false)" />

    <!-- other args -->
  <arg name="standalone" default="true" />
  <arg name="debug" default="false" />

  <arg     if="$(eval arg('standalone') or arg('debug'))" name="nodelet" value="standalone" />
  <arg unless="$(eval arg('standalone') or arg('debug'))" name="nodelet" value="load" />
  <arg     if="$(eval arg('standalone') or arg('debug'))" name="nodelet_manager" value="" />
  <arg unless="$(eval arg('standalone') or arg('debug'))" name="nodelet_manager" value="$(arg UAV_NAME)_nodelet_manager" />

  <arg     if="$(arg debug)" name="launch_prefix" value="debug_roslaunch" />
  <arg unless="$(arg debug)" name="launch_prefix" value="" />

  <group ns="$(arg UAV_NAME)">

    <node pkg="nodelet" type="nodelet" name="estimation_manager" args="$(arg nodelet) mrs_uav_state_estimation/EstimationManager $(arg nodelet_manager)" output="screen" launch-prefix="$(arg launch_prefix)">

      <param name="enable_profiler" value="$(arg PROFILER)" />
      <param name="uav_name" value="$(arg UAV_NAME)" />

      <rosparam file="$(find mrs_uav_state_estimation)/config/estimation_manager.yaml" command="load" />
      <rosparam file="$(find mrs_uav_state_estimation)/config/estimators.yaml" command="load" />

      <!-- Subscribers -->
      <remap from="~attitude_command_in" to="control_manager/attitude_cmd" />
      <remap from="~mavros_odom_in" to="mavros/local_position/odom" />
      <remap from="~garmin_range_in" to="garmin/range" />

      <!-- Publishers -->
      <remap from="~profiler" to="profiler" />
      <remap from="~pose_out" to="ground_truth_pose" />

    </node>

  </group>

</launch>
